game-core/src/main/java/games/strategy/triplea/ui/mapdata/MapData.java

```diff
@@ -127,14 +122,16 @@
   private final Map<Image, List<Point>> decorations = new HashMap<>();
   private final Map<String, Image> territoryNameImages = new HashMap<>();
   private final Map<String, Image> effectImages = new HashMap<>();
-  private final ResourceLoader resourceLoader;
 
-  public MapData(final String mapNameDir) {
-    this(ResourceLoader.getMapResourceLoader(mapNameDir));
-  }
+  @Nullable private final Image vcImage;
+  @Nullable private final Image blockadeImage;
+  @Nullable private final Image errorImage;
+  @Nullable private final Image warningImage;
```

Comment:
These 4 images were loaded when called, leveraging the fact that `MapData` had a reference to a `resourceLoader`. With the `resourceLoader` reference available now only on construction, we now load these images in the constructor.